File: planix/src/main/java/com/planix/backend/entity/Habit.java
package com.planix.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

/**
 * Сущность привычки пользователя.
 */
@Entity
@Table(name = "habit")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = {"user", "habitLogs"})
public class Habit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @OneToMany(mappedBy = "habit", cascade = CascadeType.ALL)
    private List<HabitLog> habitLogs;
}

File: planix-frontend/src/types/habit.ts
export interface Habit {
  id: number;
  name: string;
  description?: string;
}

export interface HabitLog {
  id: number;
  habitId: number;
  date: string; // YYYY-MM-DD
}

export interface HabitRequest {
  name: string;
  description?: string;
}